src/impl.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * LinkedList.cpp
+ *
+ *  Created on: 16 нояб. 2017 г.
+ *      Author: pavel
+ */
+
+#include "impl.h"
+
+LinkedList::LinkedList(): _first(nullptr) {
+}
+
+LinkedList::LinkedList(std::initializer_list<int> list): _first(nullptr)
```

Comment:
should we really initialize it here?
you can use inline initialization:

class A
{
private:
    void* pointer {nullptr};
}